src/core/thread/mle_router.cpp

```diff
@@ -4448,15 +4448,61 @@ otError MleRouter::AppendChildAddresses(Message &aMessage, Child &aChild)
     return error;
 }
 
-void MleRouter::FillRouteTlv(RouteTlv &aTlv)
+void MleRouter::FillRouteTlv(RouteTlv &aTlv, Neighbor *aNeighbor)
 {
-    uint8_t routerCount = 0;
+    uint8_t     routerIdSequence = mRouterTable.GetRouterIdSequence();
+    RouterIdSet routerIdSet      = mRouterTable.GetRouterIdSet();
+    uint8_t     routerCount;
+
+    if (aNeighbor && IsActiveRouter(aNeighbor->GetRloc16()))
```

Comment:
Should we have a test to verify that this work around works?

Reply:
I have a test, but it fails the `cli-time-sync` check.
https://github.com/openthread/openthread/runs/905142732?check_suite_focus=true

I haven't had a chance to figure out why. Let me know if you have any ideas :)

Reply:
When Time Sync feature is enabled, TimeHeader IE would be appended, thus bringing extra 17 Bytes (sizeof(TimeHeaderIE) 15B + IE Termination 2 B), and causing fragmentation.
https://github.com/openthread/openthread/blob/6488b25484aeed042a72962d1b3e3b5f2074f688/src/core/thread/mesh_forwarder.cpp#L570-L572

One possible option is when Time Sync is enabled, define a smaller `kLinkAcceptMaxRouters` (verified that value 3 could work, fyi)